src/precompiles/modexp.rs

```diff
@@ -3,87 +3,46 @@ use crate::precompiles::{
 };
 use crate::prelude::{PhantomData, Vec, U256};
 use evm::{Context, ExitError};
-use num::BigUint;
+use num::{BigUint, Integer};
 
 pub(super) const ADDRESS: [u8; 20] = super::make_address(0, 5);
 
 pub(super) struct ModExp<HF: HardFork>(PhantomData<HF>);
 
-impl ModExp<Byzantium> {
-    fn adj_exp_len(exp_len: U256, base_len: U256, bytes: &[u8]) -> U256 {
-        let mut exp32_bytes = Vec::with_capacity(32);
-        for i in 0..32 {
+impl<HF: HardFork> ModExp<HF> {
+    fn calc_iter_count(exp_len: u64, base_len: u64, bytes: &[u8]) -> u64 {
+        let mut exp_bytes = Vec::with_capacity(32);
+        for i in 0..exp_len {
```

Comment:
The whole point of this `for` loop is to extract the exponent from the input data right? Maybe worth extracting this logic into a separate function so that we only need to parse the exponent once (note we need the exponent again in `run_inner`).

Reply:
Honestly, I'm not even quite too happy with how it is. I'm going to revisit this.

Reply:
This was resolved by using slices. I'm not sure why I hadn't just done this in the first place. Might've been a reason before, but whatever it was, no longer is there.